documentation/express-route-service/express-route-gateway-bypass.md

```diff
@@ -138,92 +135,29 @@ typedef enum _sai_direction_lookup_entry_attr_t {
-
-The reverse tunnel table will have the following attributes that is common to all next hops:
-
-| Attribute name | Type | Description |
-| --- | --- | --- |
-| SAI_DASH_REVERSE_TUNNEL_ATTR_DASH_ENCAPSULATION | sai_dash_encapsulation_t | Encapsulation type, such as VxLan, NvGRE. Optional. If not specified, the encap from tunnel will be used. |
-| SAI_DASH_REVERSE_TUNNEL_ATTR_TUNNEL_KEY | sai_uint32_t | Tunnel key used in the encap, e.g. VNI in VxLAN. |
-
-A reverse tunnel supports multiple destination IPs as an ECMP group, the reverse tunnel member and reverse tunnel next hop objects will be used to specify these information:
-
-- The reverse tunnel next hop object defines the tunnel information for each destination:
-
-  | Attribute name | Type | Description |
-  | --- | --- | --- |
-  | SAI_DASH_REVERSE_TUNNEL_NEXT_HOP_ATTR_DIP | sai_ip_address_t | Destination IP used in tunnel. |
-  | SAI_DASH_REVERSE_TUNNEL_NEXT_HOP_ATTR_SIP | sai_ip_address_t | Source IP used in tunnel. |
-
-- The reverse tunnel member defines the bindings between tunnel and next hop:
-
-  | Attribute name | Type | Description |
-  | --- | --- | --- |
-  | SAI_DASH_REVERSE_TUNNEL_MEMBER_ATTR_TUNNEL_ID | sai_object_id_t | Tunnel Id |
-  | SAI_DASH_REVERSE_TUNNEL_MEMBER_ATTR_TUNNEL_NEXT_HOP_ID | sai_object_id_t | Tunnel next hop id |
+| SAI_ENI_ATTR_REVERSE_TUNNEL_SIP | sai_ip_address_t | Source IP used in the reverse tunnel. |
 
 #### 5.2.3. MSEE failover handling using flow resimulation
 
 ##### 5.2.3.1. Reverse tunnel updates
 
-When MSEE failover, the tunnel configuration will be updated. Since the tunnel is route based, we can leverage the existing [flow resimulation APIs](../dataplane/dash-flow-resimulation.md) to update the nexthop list.
+When MSEE failover, it could impact all flows that are created. In this case, we will leverage the full flow resimulation API in the existing [flow resimulation APIs](../dataplane/dash-flow-resimulation.md) to help us keep the reverse tunnel updated.
 
-Whenever the reverse tunnel id is set, we consider the reverse tunnel routing action bit is set in the pipeline, this allows us to control the flow resimulation behavior using the same mechanism.
+Following the flow resimulation design, all flow resimulation request will be explicitly requested by the SDN controller. Hence, when MSEE failover, the SDN controller will request a full flow resimulation that marks all flows as resimulation needed.
 
-1. Step 1: Update the DIP list in the reverse tunnel object, which will cover all the new flows.
-2. Step 2: Request a full flow resimulation, which will cover the existing flows.
+When the next packet from MSEE comes in for a flow, the flow will be resimulated and the reverse tunnel will be updated accordingly, by picking up the new MSEE tunnel information.
```

Comment:
What if the next packet is from the opposite side of the connection?

Reply:
it is expected to forward the packet according to the non-updated flow.

let me see if this is missed in the doc.

Reply:
found it! this behavior is captured in the section 5.2.3.3. below.

Reply:
The packet is already in slow path because the flow it belongs to was marked for resimulation. Slow path doesn't know what the old destination was.

Reply:
The old destination should be known, because it is in the existing flow.